File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        moves = 0
        nums.sort()
        
        for i in range(len(nums)-1):
            if nums[i+1] <= nums[i]:
                moves += nums[i] - nums[i+1] + 1
            nums[i+1] = max(nums[i+1], nums[i]+1)
        return moves

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minIncrementForUnique = function(nums) {
    let moves = 0;
    nums.sort((a,b) => a - b)
    
    let cnt = 0;
    for (let i = 1; i < nums.length; ++i) {
        cnt += nums[i] > nums[i - 1] ? 0 : nums[i - 1] + 1 - nums[i];
        nums[i] = Math.max(nums[i], nums[i - 1] + 1);
    }
  return cnt;
};